go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -986,23 +960,31 @@ func (sess *Session) AddDB(ctx *sql.Context, dbState InitialDbState) error {
 	return nil
 }
 
-// CreateTemporaryTablesRoot creates an empty root value and a table edit session for the purposes of storing
-// temporary tables. This should only be used on demand. That is only when a temporary table is created should we
-// create the root map and edit session map.
-func (sess *Session) CreateTemporaryTablesRoot(ctx *sql.Context, dbName string, ddb *doltdb.DoltDB) error {
-	newRoot, err := doltdb.EmptyRootValue(ctx, ddb.ValueReadWriter())
-	if err != nil {
-		return err
+func (sess *Session) AddTemporaryTable(ctx *sql.Context, tbl sql.Table) {
+	sess.tmpTables = append(sess.tmpTables, tbl)
+}
+
+func (sess *Session) DropTemporaryTable(ctx *sql.Context, name string) {
+	for i, tbl := range sess.tmpTables {
+		if strings.ToLower(tbl.Name()) == strings.ToLower(name) {
+			sess.tmpTables = append(sess.tmpTables[:i], sess.tmpTables[i+1:]...)
+			break
+		}
 	}
+}
 
-	dbState, _, err := sess.LookupDbState(ctx, dbName)
-	if err != nil {
-		return err
+func (sess *Session) GetTemporaryTable(ctx *sql.Context, name string) (sql.Table, bool) {
```

Comment:
According to the MySQL [docs](https://dev.mysql.com/doc/refman/8.0/en/create-temporary-table.html):
"TEMPORARY tables have a very loose relationship with databases (schemas)."

Should we add a database name param and check it?

Reply:
MySQL does, we should too